### PR TITLE
Fix LQ version checking out in LK-CPU tests

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
After splitting the workflow into build wheels / test wheels, LQ is not properly checked out to stable/release anymore.
 
**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-63107]

### CODE REVIEW

File: .github/workflows/tests_lkcpu_python.yml [hunk 1]
```diff
@@ -168,6 +168,22 @@ jobs:
...
+          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
+
+      - name: Switch to stable build of Lightning
+        if: inputs.lightning-version == 'stable'
+        run: |
+          cd main
+          git fetch --tags --force
+          git checkout $(git tag | sort -V | tail -1)
+          git log -1 --format='%H'
+          git status
```
Comment (maliasadi) [inline .github/workflows/tests_lkcpu_python.yml:185]: Would you like to simply install the the stable version from PyPI?

Comment (vincentmr) [reply .github/workflows/tests_lkcpu_python.yml:185]: Yes, but there is `SKIP_COMPILATION=True python -m pip install . -vv` under, which would overwrite it. So either that, or we introduce some more logic down there.

**Comment by vincentmr** [comment]
Blocked by [#5683](https://github.com/PennyLaneAI/pennylane/pull/5683).

**Review by maliasadi — APPROVED** [review]
Thanks @vincentmr!

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -37,6 +37,9 @@
 
 ### Bug fixes
 
+* The `.github/workflows/tests_lkcpu_python.yml` workflow properly checks out the release or stable version of Lightning-Qubit during the test job.
```
Comment (AmintorDusko) [inline .github/CHANGELOG.md:40]:
```suggestion
* The `.github/workflows/tests_lkcpu_python.yml` workflow properly checks out the release or stable version of LightningQubit during the test job.
```

**Review by AmintorDusko — APPROVED** [review]
Thank you for that!

File: .github/CHANGELOG.md (hunk 2 above)
Comment (AmintorDusko) [reply .github/CHANGELOG.md:40]: Super nit-pick suggestion.

Comment (vincentmr) [reply .github/CHANGELOG.md:40]: I think we usually write Lightning-Qubit for the product and `LightningQubit` for the class, so I picked the former in the present case. I'm open to revisiting and making the CHANGELOG more uniform though.

Comment (AmintorDusko) [reply .github/CHANGELOG.md:40]: Yes, indeed, Disregard my comment.

**Pull request merged**